PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/webapp/WebAppConfigFormPanelBasic.java

```diff
@@ -22,39 +22,103 @@
 
 package com.microsoft.azure.toolkit.intellij.webapp;
 
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.project.ProjectManager;
 import com.intellij.ui.TitledSeparator;
+import com.microsoft.azure.management.appservice.PricingTier;
+import com.microsoft.azure.management.resources.Subscription;
+import com.microsoft.azure.management.resources.fluentcore.arm.Region;
 import com.microsoft.azure.toolkit.intellij.appservice.AppNameInput;
 import com.microsoft.azure.toolkit.intellij.appservice.artifact.ArtifactComboBox;
 import com.microsoft.azure.toolkit.intellij.appservice.platform.PlatformComboBox;
 import com.microsoft.azure.toolkit.intellij.common.AzureFormPanel;
+import com.microsoft.azure.toolkit.lib.appservice.DraftResourceGroup;
+import com.microsoft.azure.toolkit.lib.appservice.DraftServicePlan;
 import com.microsoft.azure.toolkit.lib.appservice.Platform;
 import com.microsoft.azure.toolkit.lib.common.form.AzureFormInput;
 import com.microsoft.azure.toolkit.lib.webapp.WebAppConfig;
+import com.microsoft.azuretools.core.mvp.model.AzureMvpModel;
+import com.microsoft.azuretools.core.mvp.model.webapp.AzureWebAppMvpModel;
+import lombok.SneakyThrows;
+import org.apache.commons.lang3.StringUtils;
 
 import javax.swing.*;
 import java.nio.file.Path;
+import java.text.SimpleDateFormat;
 import java.util.Arrays;
+import java.util.Date;
 import java.util.List;
 
 public class WebAppConfigFormPanelBasic extends JPanel implements AzureFormPanel<WebAppConfig> {
+    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyMMddHHmmss");
+    public static final int RG_NAME_MAX_LENGTH = 90;
+    public static final int SP_NAME_MAX_LENGTH = 40;
+
     private JPanel contentPanel;
 
     private AppNameInput textName;
     private PlatformComboBox selectorPlatform;
     private ArtifactComboBox selectorApplication;
     private TitledSeparator deploymentTitle;
     private JLabel deploymentLabel;
+    private Subscription subscription;
+
+
+    public WebAppConfigFormPanelBasic() {
+        super();
+        $$$setupUI$$$(); // tell IntelliJ to call createUIComponents() here.
+        this.init();
+    }
 
+    private void init() {
+        final String date = DATE_FORMAT.format(new Date());
+        final Project project = ProjectManager.getInstance().getOpenProjects()[0];
+        final String defaultWebAppName = String.format("app-%s-%s", project.getName(), date);
+        this.textName.setValue(defaultWebAppName);
+        final List<Subscription> items = AzureMvpModel.getInstance().getSelectedSubscriptions();
+        this.subscription = items.get(0); // select the first subscription as the default
```

Comment:
it will error if there are no subs

Reply:
if user has no subs, he/she can not come here, right?